runtime/commonMain/src/kotlinx/serialization/MapLikeSerializer.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization
+
+/**
+ * Optional interface that represents [serializer][KSerializer] for map-like structures.
+ * [T] is a type of target type for the serializer,
+ * [K] and [V] type parameters represents types of the key and value.
+ *
+ * This interface is a marker that is recommended to implement for any serializer of [StructureKind.MAP] kind.
+ * Default formats, as well as endorsed ones, recognize serializers of [MapLikeSerializer] type and
```

Comment:
```suggestion
 * Default formats, as well as endorsed ones, may recognize serializers of [MapLikeSerializer] type and
```

Recognising is not mandatory